crates/astria-sequencer/src/app.rs

```diff
@@ -99,11 +110,18 @@ impl App {
         let state_tx = Arc::try_unwrap(arc_state_tx)
             .expect("components should not retain copies of shared state");
 
+        self.has_block_just_begun = true;
         self.apply(state_tx)
     }
 
     #[instrument(name = "App:deliver_tx", skip(self))]
     pub(crate) async fn deliver_tx(&mut self, tx: &[u8]) -> Result<Vec<abci::Event>> {
+        if self.has_block_just_begun {
+            ensure!(tx.len() == 32);
```

Comment:
is there any other validation which can be done on this tx?

Reply:
unfortunately not really, however the validators already verify this value in `ProcessProposal` by reconstructing the action tree so this is fine